sound/soc/sof/ipc4-topology.c

```diff
@@ -943,6 +946,10 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 
 	sof_ipc4_widget_update_kcontrol_module_id(swidget);
 
+	/* set pipeline core mask to keep track of the core the DP module is scheduled to run on */
+	if ((fw_module->man4_module_entry.type & SOF_IPC4_MODULE_DP))
+		spipe->core_mask |= BIT(swidget->core);
```

Comment:
Why the distinction of DP modules?
Non DP modules can only be executed on core 0?

Reply:
LL modules cannot be on a core different from the pipeline core set in topology

Reply:
@plbossart I think that's correct. Let me confirm it with the firmware team

Reply:
but what would happen if you unconditionally set this mask - regargly of the type of module? This wouldn't break today and would be future-proof if LL modules have extra permissions

Conversely if LL modules CANNOT be on a different pipeline then we need some sort of error check.

Reply:
@plbossart, I was also contemplating on this. It will not hurt at all to set the same bit several times and it will for sure future proof the code and it is at the same time makes it even cleaner: we are marking the core(s) that are used by the modules on the pipeline.

Reply:
@plbossart ok i've removed the restriction to limit this to only DP modules now.